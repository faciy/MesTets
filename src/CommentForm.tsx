import {View, Text, TextInput, TouchableOpacity} from 'react-native';
import React from 'react';

const CommentForm = props => {
  return (
    <View>
      <TextInput
        testID={'input'}
        value={props.value}
        onChangeText={props.onChange}
      />
      <TouchableOpacity testID={'submit'} onPress={props.onSubmit}>
        Submit
      </TouchableOpacity>
    </View>
  );
};

export default CommentForm;
