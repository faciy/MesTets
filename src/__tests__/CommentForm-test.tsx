import React from 'react';
import CommentForm from '../CommentForm';
import {render, cleanup, fireEvent} from '@testing-library/react-native';

afterEach(cleanup);

describe('<CommentForm />', () => {
  it('rendre le contenu', () => {
    const rendered = render(<CommentForm value={'abacaba'} />).toJSON();
    expect(rendered).toMatchSnapshot();
  });

  it('devrait déclencher des événements de modification', () => {
    const onChange = jest.fn();
    const rendered = render(<CommentForm onChange={onChange} />);
    const inputComponent = rendered.getByTestId('input');

    fireEvent(inputComponent, 'changeText', 'new text');
    expect(onChange).toHaveBeenCalledWith('new text');
  });

  it('Devrait tirer sur les événements Submit', () => {
    const onSubmit = jest.fn();
    const rendered = render(<CommentForm onSubmit={onSubmit} />);
    const buttonComponent = rendered.getByTestId('submit');

    fireEvent.press(buttonComponent, 'press');
    expect(onSubmit).toHaveBeenCalled();
  });
});
