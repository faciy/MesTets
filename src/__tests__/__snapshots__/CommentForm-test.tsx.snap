// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CommentForm /> rendre le contenu 1`] = `
<View>
  <TextInput
    testID="input"
    value="abacaba"
  />
  <View
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="submit"
  >
    Submit
  </View>
</View>
`;
